packages/app/src/components/deposit/ConfirmDepositModal.tsx

```diff
@@ -0,0 +1,171 @@
+  const symbol = staker.symbol;
+  const bwLogo = `/${staker.bwLogo}`;
+  const bwTenderLogo = `/${staker.bwTenderLogo}`;
+
+  const { validationMessage } = useBalanceValidation(tokenAmount, tokenBalance, symbol);
+  const [confirmStatus, setConfirmStatus] = useState<"None" | "Waiting">("None");
+
+  // reset to initial state
+  useEffect(() => {
+    if (show === true) {
+      setConfirmStatus("None");
+    }
+  }, [show]);
+
+  const isGRT = symbol === "GRT";
+
+  return (
+    <>
+      {show && (
+        <Layer style={{ overflow: "auto" }} animation="fadeIn" onEsc={onDismiss} onClickOutside={onDismiss}>
+          <Card
+            flex={false}
+            pad={{ vertical: "medium", horizontal: "xlarge" }}
+            width="large"
+            style={{ position: "relative" }}
+          >
+            <Button
+              style={{ position: "absolute", top: 10, right: 10 }}
+              plain
+              icon={<FormClose />}
+              onClick={onDismiss}
+            />
+            <CardHeader justify="center" pad="none">
+              <Heading level={2} alignSelf="center">
+                {`Confirm Stake ${symbol} for t${symbol}`}
+              </Heading>
+            </CardHeader>
+            <CardBody>
+              <Box pad={{ top: "medium", horizontal: "large" }} align="center">
+                <Form style={{ width: "100%" }}>
+                  <Box gap="medium">
+                    <FormField label={`Stake ${symbol}`}>
+                      <TextInput
+                        id="formDepositStake"
+                        type="number"
+                        value={tokenAmount}
+                        onChange={(e) => setTokenAmount(e.target.value)}
+                        required={true}
+                        style={{ textAlign: "right", padding: "20px 50px" }}
+                        icon={
+                          <Box pad="xsmall" direction="row" align="center" gap="small">
+                            <Image height="35" src={bwLogo} />
+                            <Text>{symbol}</Text>
+                          </Box>
+                        }
+                      />
+                      <Text color="red">{validationMessage}</Text>
+                      <AmountInputFooter
+                        label={`Balance: ${weiToEthWithDecimals(tokenBalance, 6)} ${symbol}`}
+                        onClick={() => {
+                          setTokenAmount(utils.formatEther(tokenBalance.toString() ?? "0"));
+                        }}
+                      />
+                    </FormField>
+                    <FormField label={`Receive t${symbol}`}>
+                      <TextInput
+                        readOnly
+                        id="formDepositReceive"
+                        placeholder={"0"}
+                        value={isGRT ? Number.parseInt(tokenAmount) * 0.995 : tokenAmount}
+                        style={{ textAlign: "right", padding: "20px 50px" }}
```

Comment:
We can create a calcDeposit hook and use https://github.com/Tenderize/tender-core/blob/db0ec50182e46981e5912fa2187a2554b99c02bd/contracts/tenderizer/Tenderizer.sol#L193